opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/EncryptionServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package de.adorsys.opba.protocol.facade.services;
+
+import com.google.common.io.Resources;
+import com.google.crypto.tink.Aead;
+import com.google.crypto.tink.CleartextKeysetHandle;
+import com.google.crypto.tink.JsonKeysetReader;
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.aead.AeadConfig;
+import com.google.crypto.tink.subtle.AesGcmJce;
+import de.adorsys.opba.protocol.api.services.EncryptionService;
+import lombok.SneakyThrows;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.springframework.stereotype.Service;
+
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import java.nio.file.Paths;
+import java.security.Provider;
+import java.security.Security;
+import java.util.Base64;
+
+@Service
+public class EncryptionServiceImpl implements EncryptionService {
+
+    private Aead aeadSystem;
+    private Provider provider;
+
+    @SneakyThrows
+    public EncryptionServiceImpl() {
+        AeadConfig.register();
+        String path = Paths.get(Resources.getResource("keyset.json").toURI()).toAbsolutePath().toString();
+        KeysetHandle systemKeysetHandle = CleartextKeysetHandle.read(JsonKeysetReader.withPath(path));
+        aeadSystem = systemKeysetHandle.getPrimitive(Aead.class);
+
+        if (null == Security.getProperty(BouncyCastleProvider.PROVIDER_NAME)) {
```

Comment:
Meaningless check, probably you wanted 
Security.getProperty(SECURITY_PROVIDER)

Reply:
no, it checks if BC added to security or not

Reply:
What you want is the ability to use custom security provider in accordance with `PBEWithSHA256And256BitAES-CBC-BC` - it is BouncyCastle, but if someone wants JCE we should provide it.